src/main/java/com/minecraftabnormals/abnormals_core/core/util/BlockUtil.java

```diff
@@ -156,7 +157,21 @@ public static BlockState getStateAtOffsetPos(IBlockSource source) {
 	 * @return A {@link List} of entities at the at the offset position.
 	 * @see #offsetPos(IBlockSource source)
 	 */
-	public static List<Entity> getEntitiesAtOffsetPos(IBlockSource source, Class<Entity> entityType) {
+	public static List<Entity> getEntitiesAtOffsetPos(IBlockSource source, Class<? extends Entity> entityType) {
 		return source.getWorld().getEntitiesWithinAABB(entityType, new AxisAlignedBB(offsetPos(source)));
 	}
+
+	/**
+	 * Gets a {@link List} of entities that match a {@link Predicate} at the position returned by {@link #offsetPos(IBlockSource source)}.
+	 *
+	 * @param source The {@link IBlockSource} to get the position from.
+	 * @param entityType The class extending {@link Entity} to search for. Set to {@code Entity.class} to get all entities, regardless of type.
+	 * @param predicate The predicate with type parameter extending {@link Entity} to check against.
+	 *
+	 * @return A {@link List} of entities at the at the offset position.
+	 * @see #offsetPos(IBlockSource source)
+	 */
+	public static List<Entity> getEntitiesAtOffsetPos(IBlockSource source, Class<? extends Entity> entityType, Predicate<? super Entity> predicate) {
```

Comment:
This method doesn't exactly do what the documentation says it does.
`The class extending {@link Entity} to search for.`
In this case, the predicate and list should both be typed with the same entity as the `entityType` parameter is.

Reply:
Hmm, I ended up changing the generics to be the same as `World`'s methods, but the javadocs are probably still not clear enough.